functests/ptp/ptp.go

```diff
@@ -108,4 +93,417 @@ var _ = Describe("ptp", func() {
+				fmt.Sprint("PTP Nodes with lable %s are not deployed on cluster", ptpSlaveNodeLabel))
+
+			ptpConfigTest := mutateProfile(ptpConfigSlave, ptpConfigName, nodes.Items[0].Name)
+
+			status := client.Client.ConfigV1Interface.RESTClient().Post().AbsPath(configPtpAPIPath).
+				Resource("ptpconfigs").Body(ptpConfigTest).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP config creation Error"))
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			testPtpPod, err = replaceTestPod(testPtpPod, 60)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		var _ = Context("Check if Node has Profile", func() {
+			//25743
+			It("PTP daemon can apply match rule based on nodeLabel", func() {
+				waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: test")
+			})
+		})
+		AfterEach(func() {
+			status := client.Client.ConfigV1Interface.RESTClient().Delete().AbsPath(configPtpAPIPath + "/test").Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("Can not delete PTP config"))
+
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP slave config recovery Error"))
+			waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: slave")
+		})
+	})
 })
+
+func podRole(runningPod v1core.Pod, role string) bool {
+	nodeList, err := client.Client.Nodes().List(metav1.ListOptions{
+		LabelSelector: role,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	for NodeNumber := range nodeList.Items {
+		if runningPod.Spec.NodeName == nodeList.Items[NodeNumber].Name {
+			return true
+		}
+	}
+	return false
+}
+
+func compareOffsetTime(timeDiff string) bool {
+	var timeStampList []int
+	for _, line := range strings.Split(timeDiff, "\n") {
+		if strings.Contains(line, "openshift_ptp_max_offset_from_master") && !strings.Contains(line, "# ") {
+			lineValues := strings.Split(line, " ")
+			lastValue := strings.Trim(lineValues[len(lineValues)-1], "\r")
+			offsetFromMaster, err := strconv.Atoi(lastValue)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(offsetFromMaster).To(BeNumerically("<", 100))
+			Expect(offsetFromMaster).To(BeNumerically(">", -100))
+			timeStampList = append(timeStampList, offsetFromMaster)
+		}
+	}
+	Expect(len(timeStampList)).To(BeNumerically("==", 2))
+	return true
+}
+
+func getPodLogs(pod v1core.Pod) (string, error) {
+	podLogOpts := v1core.PodLogOptions{}
+	req := client.Client.CoreV1Interface.Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream()
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+
+	return str, nil
+}
+
+// This function parses ethtool command output and detect interfaces which supports ptp protocol
+func netParseEthToolTimeStampFeature(cmdOut *bytes.Buffer) bool {
```

Comment:
Naming this is hard :-) Can it be something like `isPTPEnabled` and have the `cmdOut` named `ethToolOutput` ?